### PR TITLE
CM-590: How to Implement LinkedList in C#

### PR DESCRIPTION
**Pull request by FilipiBraga**
LinkedList and CustomLinkedList implementation

### CODE REVIEW

**Review by PhilBroderick — CHANGES_REQUESTED** [review]
Looks fine. There'll be some changes required that are noted in the feedback document.

**Review by PhilBroderick — APPROVED** [review]

File: collections-lists/LinkedList/LinkedList/CustomImplementation/CustomLinkedList.cs [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+{
+    public class CustomLinkedList<T> : IEnumerable<Node<T>>
+    {
+        public Node<T>? First { get; set; }
+        public int Count { get; set; }
+
+        public void AddFirst(Node<T> nodeToAdd)
+        {
+            if (First is null)
+                First = nodeToAdd;
```
Comment (PhilBroderick) [inline collections-lists/LinkedList/LinkedList/CustomImplementation/CustomLinkedList.cs:19]: Could we add brackets for this `if` statement? The article has already been changed.

**Review by PhilBroderick — CHANGES_REQUESTED** [review]
Just one small change to make to the code. Also if we can move this solution to the `collections-csharp` folder instead as the `collections-lists` folder is strictly related to problems around the `List` class.

File: collections-lists/LinkedList/LinkedList/CustomImplementation/CustomLinkedList.cs (hunk 1 above)
Comment (FilipiBraga) [reply collections-lists/LinkedList/LinkedList/CustomImplementation/CustomLinkedList.cs]: I have moved to the correct folder and added brackets to a couple of `if` statements that were missing.

**Review by PhilBroderick — APPROVED** [review]

**Pull request merged**